builtin/credential/oidc/path_config.go

```diff
@@ -0,0 +1,275 @@
+}
+
+func (b *backend) pathConfigRead(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+
+	cfg, err := b.Config(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	if cfg == nil {
+		return nil, nil
+	}
+
+	resp := &logical.Response{
+		Data: map[string]interface{}{
+			"issuer_url":       cfg.IssuerUrl,
+			"issuer_verify_ca": cfg.IssuerCABundle,
+			"client_ids":       cfg.ClientIDs,
+			"username_claim":   cfg.UsernameClaim,
+			"groups_claim":     cfg.GroupsClaim,
+		},
+	}
+
+	return resp, nil
+}
+
+func (b *backend) pathConfigWrite(
+	req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	cfg, err := b.Config(req.Storage)
+	if err != nil {
+		return nil, err
+	}
+	// Due to the existence check, entry will only be nil if it's a create
+	// operation, so just create a new one
+	if cfg == nil {
+		cfg = &ConfigEntry{}
+	}
+
+	// Parse issuer_url, required.
+	issuerUrlRaw, ok, err := d.GetOkErr("issuer_url")
+	if ok {
+		_, err = url.Parse(issuerUrlRaw.(string))
+		if err != nil {
+			return logical.ErrorResponse(fmt.Sprintf("Error parsing given issuer_url: %s", err)), nil
+		}
+		cfg.IssuerUrl = issuerUrlRaw.(string)
+	} else if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	} else {
+		return logical.ErrorResponse("auth/oidc: missing required config value issuer_url"), nil
+	}
+	// Parse issuer_verify_ca.
+	issuerVerifyCaRaw, ok, err := d.GetOkErr("issuer_verify_ca")
+	if ok {
+		_, err := certPoolFromString(issuerVerifyCaRaw.(string))
+		if err != nil {
+			return logical.ErrorResponse("auth/oidc: failed parsing PEM of issuer_verify_ca: " + err.Error()), nil
+		}
+		cfg.IssuerCABundle = issuerVerifyCaRaw.(string)
+	} else if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+	// Parse client ids, required.
+	clientIdsRaw, ok, err := d.GetOkErr("client_ids")
+	if ok {
+		cfg.ClientIDs = clientIdsRaw.([]string)
+	} else if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	} else if !ok {
+		return logical.ErrorResponse("auth/oidc: missing required config value client_ids"), nil
+	}
+	// Read defaults for claim configs.
+	cfg.UsernameClaim = d.Get("username_claim").(string)
+	cfg.GroupsClaim = d.Get("groups_claim").(string)
+
+	jsonCfg, err := logical.StorageEntryJSON("config", cfg)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+	if err := req.Storage.Put(jsonCfg); err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	return nil, nil
+}
+
+func (b *backend) pathConfigExistenceCheck(
+	req *logical.Request, d *framework.FieldData) (bool, error) {
+	cfg, err := b.Config(req.Storage)
+	if err != nil {
+		return false, err
+	}
+
+	return cfg != nil, nil
+}
+
+func (b *backend) oidcProviderForConfig(conf *ConfigEntry) (*oidc.Provider, error) {
+	// TODO(mwitkow): This creates a new Provider each time, making a request to the Discovery URL of the provider. Add caching per config "hash".
```

Comment:
This should probably just be a field on the backend. As noted, it's fairly expensive to create a provider.